From c26bd138f84d8b18ad3f17c20647e5045e68f1c2 Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Mon, 6 Dec 2021 08:51:48 +0000
Subject: [PATCH] Use more system libraries

---
 cmake/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index acbde7f56..49f6ceaa1 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -741,11 +741,11 @@ if (Protobuf_FOUND)
     set(PROTOBUF_LIB protobuf::libprotobuf)
     #We have a check here but most of the cmake users don't know the Protobuf_USE_STATIC_LIBS
     # variable exists and may leave it in a wrong state.
-    if (NOT Protobuf_USE_STATIC_LIBS)
+    #if (NOT Protobuf_USE_STATIC_LIBS)
       # ONNX Runtime itself can work in such a setting but it may cause compatibility issues
       # when ONNX Runtime is integrated with the other ONNX ecosystem softwares.
-      message(WARNING "Use Protobuf_USE_STATIC_LIBS to ensure compatibility with other ONNX ecosystem components.")
-    endif()
+    #  message(WARNING "Use Protobuf_USE_STATIC_LIBS to ensure compatibility with other ONNX ecosystem components.")
+    #endif()
   else()
     set(PROTOBUF_LIB protobuf::libprotobuf-lite)
   endif()
-- 
2.32.0

