From 9b427ec13465c7f472d6bd45d7b072e1da218d94 Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Mon, 19 Jun 2023 11:17:28 +0000
Subject: [PATCH] Fix shared lib re2 not found

---
 cmake/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 2e54063598..200e39204f 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -1491,6 +1491,7 @@ if (WIN32)
   list(APPEND onnxruntime_EXTERNAL_LIBRARIES debug Dbghelp)
 else()
   list(APPEND onnxruntime_EXTERNAL_LIBRARIES nsync::nsync_cpp)
+  list(APPEND onnxruntime_EXTERNAL_LIBRARIES re2)
   list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${ICONV_LIB} ${CMAKE_DL_LIBS} Threads::Threads)
 endif()
 
-- 
2.34.1

