From d30b7a48e0a60bb04202724087ba7cb4075484e8 Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Mon, 21 Aug 2023 15:22:44 +0000
Subject: [PATCH] Pin numpy to open-ce's version

---
 requirements.txt.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requirements.txt.in b/requirements.txt.in
index 89242061fb..a5d47e25a4 100644
--- a/requirements.txt.in
+++ b/requirements.txt.in
@@ -1,6 +1,6 @@
 coloredlogs
 flatbuffers
-numpy >= @Python_NumPy_VERSION@
+numpy >= 1.24.3
 packaging
 protobuf
 sympy
-- 
2.34.1

